app/indexer/settling/BastionIndexer.ts

```diff
@@ -16,36 +16,65 @@ export default class BastionIndexer extends BaseContractIndexer {
   }
 
   async bastionArmyMoved(event: Event): Promise<void> {
+    const eventId = event.eventId as string;
     const params = event.parameters;
-    const lon = BigNumber.from(params[0]).toNumber();
-    const lat = BigNumber.from(params[1]).toNumber();
+    const longitude = BigNumber.from(params[0]).toNumber();
+    const latitude = BigNumber.from(params[1]).toNumber();
     const bastionPastLocation = parseInt(params[2]);
     const bastionCurrentLocation = parseInt(params[3]);
-    const realmId = arrayUInt256ToNumber(params.slice(4, 6));
-    const armyId = parseInt(params[6]);
-    const bastionId = createBastionId(lat, lon);
+    const bastionArrivalBlock = parseInt(params[4]);
+    const realmId = arrayUInt256ToNumber(params.slice(5, 7));
+    const armyId = parseInt(params[7]);
+    const bastionId = createBastionId(latitude, longitude);
```

Comment:
previously, we'd update the event name so we didn't have to worry about change in event signatures. early events had fewer parameters, so you'll have to test to make sure it doesn't create any invalid db entries. since the parameter size is different you may want to validate that and return early so we don't process earlier events.

Reply:
good point. What was happening currently is that the both army update and saveRealmHistory promises were failing because realmId and armyId were not right, so it's like it was not indexing these events. Ive changed the code so that it filters them out directly to make it more clear we dont want these. 
Fortunately all the armies linked to these events were killed in the bastion already so its fine to filter these events.